Observer/UserProfileNewsletterSubscribeObserver.php

```diff
@@ -12,32 +12,36 @@ class UserProfileNewsletterSubscribeObserver implements ObserverInterface
 
     public function __construct(
         \Klaviyo\Reclaim\Helper\Data $data_helper,
-        \Magento\Framework\App\RequestInterface $request,
         \Magento\Customer\Api\CustomerRepositoryInterface $customer_repository_interface
     ) {
-        $this->data_helper = $data_helper;
-        $this->request = $request;
+        $this->data_helper                   = $data_helper;
         $this->customer_repository_interface = $customer_repository_interface;
     }
 
     public function execute(\Magento\Framework\Event\Observer $observer)
     {
-        if (!$this->data_helper->getEnabled()) return;
-
+        if (!$this->data_helper->getEnabled()) {
+            return;
+        }
+        $customer   = null;
         $subscriber = $observer->getDataObject();
 
         if ($subscriber->isStatusChanged()) {
-          $customer = $this->customer_repository_interface->getById($subscriber->getCustomerId());
-
-          if ($subscriber->isSubscribed()) {
-            $this->data_helper->subscribeEmailToKlaviyoList(
-                $customer->getEmail(),
-                $customer->getFirstname(),
-                $customer->getLastname()
-            );
-          } else {
-            $this->data_helper->unsubscribeEmailFromKlaviyoList($customer->getEmail());
-          }
+            if ($subscriber->getCustomerId()) {
+                $customer = $this->customer_repository_interface->getById($subscriber->getCustomerId());
+            }
+
+            if ($subscriber->isSubscribed()) {
+                $this->data_helper->subscribeEmailToKlaviyoList(
```

Comment:
could we implement below as:

```
if($customer)
   $subscriber = $customer;
```

and then the below would just be:

```
$this->data_helper->subscribeEmailToKlaviyoList(
                $subscriber->getEmail(),
                $subscriber->getFirstname(),
                $subscriber->getLastname()
            );
```

Reply:
`subscriber` and `customer`, are different objects and it will be not right using it as the same